src/main/java/de/fraunhofer/aisec/cpg/helpers/Util.java

```diff
@@ -79,6 +79,36 @@
         .collect(Collectors.toList());
   }
 
+  /**
+   * Returns the first element of the specified Class-type {@param specifiedClass} that has the name
+   * {@param name} in the list {@param listOfNodes}.
+   *
+   * @param <S> Some class that extends {@link Node}.
+   */
+  public static <S extends Node> S getOfTypeWithName(
```

Comment:
Btw it seems that a similar function already exists in `Util.subnodesOfType`

Reply:
@oxisto Is that relevant here? This function specifically should work not only on TranslationUnits

Reply:
I am referring to those functions
https://github.com/Fraunhofer-AISEC/cpg/blob/1024f02164d547f961c0f2f3a6d0642410ef6711/src/main/java/de/fraunhofer/aisec/cpg/helpers/Util.java#L91-L104

There is an exponential increase of utility functions and we should be a bit careful here.

Reply:
I would recommend such a workflow instead, using TestUtils functions created specifically for this purpose: https://github.com/Fraunhofer-AISEC/cpg/blob/64ca4c77f23cc25d41eada7524d7db29f2ab5739/src/test/java/de/fraunhofer/aisec/cpg/enhancements/VariableResolverTest.java#L50-L51

Reply:
Good point, @Masrepus. Can we document this somehow, i.e. in a `CONTRIBUTING.md` or something? At "some" point we probably want to cleanup the unit tests, but at least it would be good to have a a reference for future tests.

Reply:
Yeah, cleaning up the tests and unifying them is something I'm currently doing (or better will continue to be doing after the exams)

Reply:
> I would recommend such a workflow instead, using TestUtils functions created specifically for this purpose:
> 
> https://github.com/Fraunhofer-AISEC/cpg/blob/64ca4c77f23cc25d41eada7524d7db29f2ab5739/src/test/java/de/fraunhofer/aisec/cpg/enhancements/VariableResolverTest.java#L50-L51

I don't like that idea because it works on the elements twice, which is not a large issue when we talk about testing, but specifically because we then have to add an additional line for that call in 28 + 17 places in those two TestClasses only, at that point I prefer single utility function more, to reduce code reuse. I will however see which functions in Utils are only used in tests and extract them to the TestUtils class.

Reply:
> @oxisto Is that relevant here? This function specifically should work not only on TranslationUnits

Ah the duplicate is one message that I did not see when replying

Reply:
I mean you can of course create a function that wraps the subnode and finding calls into one method, but inside this function you should probably use calls to the existing subnode and finding methods, as your method basically recreates everything the other two are doing already, only in a single method

Reply:
Additionally the assumption that exactly one node matching the criteria needs to be present in some cases is enforced in `findByUniqueName` instead of just returning null

Reply:
Partially true, thats why I mentioned that it will increase the runtime of every place where these functions are used, however, they are only used in tests where this is ok, I will see if I can either use a customizable predicate(checking for name and/or class and therefore reducing runtime) there or adopt your proposal.

Reply:
There is also a finding function that takes a custom predicate

Reply:
Although I ment one that does not need another call but merges to of my functions to use only predicates and performs only one run

Reply:
Also these functions perform not only finding by predicate but specifically for the find by type name, also cast the returning list to the search type, not the type of the given collection. I will leave it by moving the functions. If someone finds a way to reduce the number of functions without increasing the code in the test files by a to large portion he is welcome to do so.